media-sound/tenacity/tenacity-1.3~beta.recipe

```diff
@@ -0,0 +1,145 @@
+	lib:libatomic$secondaryArchSuffix
+	lib:libavcodec$secondaryArchSuffix
+	lib:libavformat$secondaryArchSuffix
+	lib:libavutil$secondaryArchSuffix
+	lib:libcairo$secondaryArchSuffix
+	lib:libexpat$secondaryArchSuffix
+	lib:libflac$secondaryArchSuffix
+	lib:libgdk_pixbuf_2.0$secondaryArchSuffix
+	lib:libglib_2.0$secondaryArchSuffix
+	lib:libgtk_3$secondaryArchSuffix
+	lib:libharfbuzz$secondaryArchSuffix
+	lib:libiconv$secondaryArchSuffix
+	lib:libid3tag$secondaryArchSuffix
+	lib:libintl$secondaryArchSuffix
+	lib:libmad$secondaryArchSuffix
+	lib:libmp3lame$secondaryArchSuffix
+	lib:libogg$secondaryArchSuffix
+	lib:libopus$secondaryArchSuffix
+	lib:libpango_1.0$secondaryArchSuffix
+	lib:libportaudio$secondaryArchSuffix
+	lib:libportmidi$secondaryArchSuffix
+	lib:libportsmf$secondaryArchSuffix
+	lib:libsndfile$secondaryArchSuffix
+	lib:libsoundtouch$secondaryArchSuffix
+	lib:libsoxr$secondaryArchSuffix
+	lib:libsqlite3$secondaryArchSuffix
+	lib:libtwolame$secondaryArchSuffix
+	lib:libuuid$secondaryArchSuffix
+	lib:libvorbis$secondaryArchSuffix
+	lib:libX11$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	wxgtk${secondaryArchSuffix}_devel
+	devel:libavcodec$secondaryArchSuffix
+	devel:libavformat$secondaryArchSuffix
+	devel:libavutil$secondaryArchSuffix
+	devel:libexpat$secondaryArchSuffix
+	devel:libflac$secondaryArchSuffix
+	devel:libglib_2.0$secondaryArchSuffix
+	devel:libgtk_3$secondaryArchSuffix
+	devel:libiconv$secondaryArchSuffix
+	devel:libid3tag$secondaryArchSuffix
+	devel:libintl$secondaryArchSuffix
+	devel:libmad$secondaryArchSuffix
+	devel:libmp3lame$secondaryArchSuffix
+	devel:libogg$secondaryArchSuffix
+	devel:libopus$secondaryArchSuffix
+	devel:libportaudio$secondaryArchSuffix
+	devel:libportmidi$secondaryArchSuffix
+	devel:libportsmf$secondaryArchSuffix
+	#devel:libpostproc$secondaryArchSuffix
+	devel:libsndfile$secondaryArchSuffix
+	devel:libsoundtouch$secondaryArchSuffix
+	devel:libsoxr$secondaryArchSuffix
+	devel:libsqlite3$secondaryArchSuffix
+	devel:libtwolame$secondaryArchSuffix
+	devel:libuuid$secondaryArchSuffix
+	devel:libvorbis$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:awk
+	cmd:cmake
+	cmd:diff
+	cmd:find
+	cmd:gcc$secondaryArchSuffix
+	cmd:gettext$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:ninja
+	cmd:pkg_config$secondaryArchSuffix
+	cmd:which
+	cmd:wx_config$secondaryArchSuffix
+	"
+
+defineDebugInfoPackage tenacity$secondaryArchSuffix \
+	$binDir/tenacity$secondaryArchSuffix
+
+BUILD()
+{
+	mkdir -p build/src/private
+	cmake -S . -B build -G Ninja \
+		$cmakeDirArgs \
+		-DCMAKE_BUILD_TYPE=Release \
+		-DAUDACITY_REV_LONG="$srcGitRev" \
+		-DAUDACITY_REV_TIME="$srcGitDate" \
+		-DPCH=no
+		#-Duse_vst=off
+	cmake --build build $jobArgs
+}
+
+INSTALL()
+{
+	cmake --install build
+
+	mkdir -p $appsDir
+	ln -s $binDir/tenacity $appsDir/Tenacity
+
+	local APP_SIGNATURE="application/x-vnd.tenacity"
```

Comment:
Unless if this is explicitly Haiku-specific (sorry, bit rusty), the signature should likely be different (something like `org.tenacityaudio.Tenacity`).

Reply:
it has to start with "application/x-vnd." because it's a MIME type. Since "x-" means "this is nonstandard" and "vnd" means "this is vendor specific", there are unfortunately no rules to how this namespace is used. It's usually "vendor name.app name" or something like that.

Not that it really matters, as long as it is unique, it should be fine.

Reply:
Yeah, was aware about `"application/x-vnd.", it's the capitalization that kind bothered me and I have had some issues specifically related to the application signature before -- they mostly concerned plugins and creating new windows in conjuction with Audacity, but it's safe to presume that an Audacity port will not contain such a conflict.

I was just trying to double check as I had some problems with creating new windows -- or that this signature is *Haiku-specific* in the sense that it does not affect the functionality of the application (which it shouldn't, as it gets baked into the resource file). I was worried as I had some issues in the past as I was testing OS-wide default applications when using Tenacity at the same time as Audacity on all platforms, but this should not be the case here because Haiku does things nicer.

Reply:
Anyways, I'm probably not making much sense and I glossed over this PR, please do not consider my comments and my attempts at giving extra context as some sort of a blocker.